src/mock/quick_p2p/network.rs

```diff
@@ -218,13 +216,31 @@ impl Queue {
         self.0.push_back(packet)
     }
 
-    fn pop(&mut self) -> Option<Packet> {
-        self.0.pop_front()
-    }
-
     fn is_empty(&self) -> bool {
         self.0.is_empty()
     }
+
+    // This function will pop random msg from the queue.
+    fn pop_random_msg(&mut self, rng: &mut SeededRng) -> Option<Packet> {
+        let first_non_random_msg = self
```

Comment:
This variable is weirdly named. I'd call it something like `first_non_msg_packet` (that is: the first packet that is not `Message`).